### PR TITLE
GaussianNLLLoss no_batch_dim docs and testing

### PR DESCRIPTION
**Pull request by george-qi**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #69783

Differential Revision: [D33200486](https://our.internmc.facebook.com/intern/diff/D33200486)

### CODE REVIEW

**Comment by george-qi** [comment]
@george-qi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33200486).

File: torch/testing/_internal/common_modules.py [hunk 1]
```diff
@@ -227,6 +227,31 @@ def reference_fn(m, p, i, t, constructor_kwargs=constructor_kwargs):
...
+        ('reduction_none', {'reduction': 'none'}),
+    ]
+
+    module_inputs = []
+    for desc, constructor_kwargs in cases:
+        module_inputs.append(
+            ModuleInput(constructor_input=FunctionInput(**constructor_kwargs),
+                        forward_input=FunctionInput(make_input((5, 3)),
+                                                    make_target((5, 3)),
+                                                    make_input((5, 1)).abs()),
```
Comment (jbschlosser) [inline torch/testing/_internal/common_modules.py:247]: I think these input shapes have batch dims, no?

**Review by jbschlosser — COMMENTED** [review]
One small testing question but otherwise looks nice

**Comment by george-qi** [comment]
@george-qi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33200486).

**Review by jbschlosser — APPROVED** [review]
LGTM!

**Pull request closed without merging**